### PR TITLE
suggestion: xkcd stdcommand

### PR DESCRIPTION
**Pull request by mrbentarikau**
I hope this fits to general bot's narrative... errors and embed part could be a bit wrong.
Currently runs fine, but I don't know all the nooks and crannies of the bot yet : )

gives random xkcd if no arguments given, optional arg is number and flag -l gives latest comic

### CODE REVIEW

File: stdcommands/xkcd/xkcd.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		embed := &discordgo.MessageEmbed{
+			Title:       fmt.Sprintf("#%d: %s", xkcd.Num, xkcd.Title),
+			Description: fmt.Sprintf("[%s](%s%d/)", xkcd.Alt, XkcdHost, xkcd.Num),
+			Color:       int(rand.Int63n(16777215)),
+			Image: &discordgo.MessageEmbedImage{
+				URL: xkcd.Img,
+			},
+		}
+
+		_, errEmbed := common.BotSession.ChannelMessageSendEmbed(data.Msg.ChannelID, embed)
```
Comment (jonas747) [inline stdcommands/xkcd/xkcd.go]: You can just return the embed, no need to send it manually

File: stdcommands/xkcd/xkcd.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			if n >= 1 && n <= xkcd.Num {
+				xkcdNum = n
+			} else {
+				return fmt.Sprintf("There's no comic numbered %d, current range is 1-%d", n, xkcd.Num), nil
+			}
+		}
+
+		//if no latest flag is set, fetches a comic by number
+		if !latest {
+			queryBody, err = getComic(xkcdNum)
```
Comment (jonas747) [inline stdcommands/xkcd/xkcd.go]: Doing it using this method you have to call getComic twice, which is a bit unnecessary

File: stdcommands/xkcd/xkcd.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, true
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, true
+	}
```
Comment (jonas747) [inline stdcommands/xkcd/xkcd.go]: Redcude duplicated code by also handling json umarshalling here

File: stdcommands/xkcd/xkcd.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+		_, errEmbed := common.BotSession.ChannelMessageSendEmbed(data.Msg.ChannelID, embed)
+		if errEmbed != nil {
+			return errEmbed, errEmbed
+		}
+		return nil, nil
+	},
+}
+
+func getComic(number ...int64) ([]byte, bool) {
```
Comment (jonas747) [inline stdcommands/xkcd/xkcd.go]: Return an error instead of boolean to be idiomatic

Comment (mrbentarikau) [reply stdcommands/xkcd/xkcd.go]: updated whole code

File: stdcommands/xkcd/xkcd.go (hunk 3 above)
Comment (mrbentarikau) [reply stdcommands/xkcd/xkcd.go]: updated whole code below

**Comment by jonas747** [comment]
Lokos good

**Pull request merged**